pkg/koordlet/metricsadvisor/collectors/coldmemoryresource/cold_page_kidled.go

```diff
@@ -0,0 +1,234 @@
+	"go.uber.org/atomic"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metricsadvisor/framework"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/resourceexecutor"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/statesinformer"
+	koordletutil "github.com/koordinator-sh/koordinator/pkg/koordlet/util"
+	"github.com/koordinator-sh/koordinator/pkg/util"
+)
+
+type kidledcoldPageCollector struct {
+	collectInterval time.Duration
+	started         *atomic.Bool
+	cgroupReader    resourceexecutor.CgroupReader
+	statesInformer  statesinformer.StatesInformer
+	podFilter       framework.PodFilter
+	appendableDB    metriccache.Appendable
+	metricDB        metriccache.MetricCache
+}
+
+func (k *kidledcoldPageCollector) Run(stopCh <-chan struct{}) {
+	go wait.Until(k.collectColdPageInfo, k.collectInterval, stopCh)
+}
+
+func (k *kidledcoldPageCollector) Started() bool {
+	return k.started.Load()
+}
+
+func (k *kidledcoldPageCollector) Enabled() bool {
+	return true
+}
+
+func (k *kidledcoldPageCollector) Setup(c1 *framework.Context) {}
+
+func (k *kidledcoldPageCollector) collectColdPageInfo() {
+	if k.statesInformer == nil {
+		return
+	}
+	coldPageMetrics := make([]metriccache.MetricSample, 0)
+
+	nodeColdPageInfoMetric, err := k.collectNodeColdPageInfo()
+	if err != nil {
+		klog.Warningf("generate node cold page info metrics failed, err %v", err)
+	}
+	coldPageMetrics = append(coldPageMetrics, nodeColdPageInfoMetric...)
+
+	podsColdPageInfoMetric, err := k.collectPodsColdPageInfo()
+	if err != nil {
+		klog.Warningf("generate pods or container cold page info metrics failed, err %v", err)
+	}
+	coldPageMetrics = append(coldPageMetrics, podsColdPageInfoMetric...)
+
+	appender := k.appendableDB.Appender()
+	if err := appender.Append(coldPageMetrics); err != nil {
+		klog.ErrorS(err, "Append node metrics error")
+		return
+	}
+
+	if err := appender.Commit(); err != nil {
+		klog.Warningf("Commit node metrics failed, reason: %v", err)
+		return
+	}
+
+	k.started.Store(true)
+}
+
+func (k *kidledcoldPageCollector) collectNodeColdPageInfo() ([]metriccache.MetricSample, error) {
+	coldPageMetrics := make([]metriccache.MetricSample, 0)
+	collectTime := time.Now()
+	coldPageInfo, err := k.cgroupReader.ReadMemoryIdlePageStatsByKidled("")
+	if err != nil {
+		return nil, err
+	}
+	nodeColdPageBytes := coldPageInfo.GetColdPageTotalBytes()
+	nodeColdPageBytesValue := float64(nodeColdPageBytes)
+	nodeColdPageMetrics, err := metriccache.NodeMemoryColdPageSizeMetric.GenerateSample(nil, collectTime, nodeColdPageBytesValue)
+	if err != nil {
+		return nil, err
+	}
+	coldPageMetrics = append(coldPageMetrics, nodeColdPageMetrics)
+	memInfo, err := koordletutil.GetMemInfo()
+	if err != nil {
+		return nil, err
+	}
+	memUsageWithHotPageBytes := memInfo.MemTotal*1024 - memInfo.MemFree*1024 - nodeColdPageBytes
```

Comment:
please wrap the calculation of the hot page usage into a function under the pkg koordlet/util/system